providers/MetaMaskWalletProvider.tsx

```diff
@@ -31,13 +33,17 @@ const MetaMaskWalletProvider: FC<MetaMaskWalletProviderProps> = ({
 	const [selectedAccount, setSelectedAccount] =
 		useState<MetaMaskWalletContextType["selectedAccount"]>(null);
 
+	const section = useSectionUri();
+
 	const connectWallet = useCallback(
 		async (callback) => {
 			if (!extension) {
 				callback?.();
 				return promptInstallExtension();
 			}
 
+			if (section === "bridge") await ensureEthereumChain(extension);
+
```

Comment:
I think we should move this logic to `SubmitButton` and keep `MetaMaskWalletProvider` unaware of what section it's in